CodeU_Assignement_5/src/codeu_assignement_5/KemUnknownLanguage.java

```diff
@@ -0,0 +1,145 @@
+package codeu_assignement_5;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author Karine
+ */
+public class KemUnknownLanguage {
+    /*
+     * Input: a dictionary (a list of words in lexicographic order) of all words in an unknown/invented language, 
+    *  Output: alphabet (an ordered list of characters) of that language, letters we cannot decide are appended in the end.
+    */
+    public List<Character> getAlphabeit(String[] alienWords) {
+        if ((alienWords != null) && (alienWords.length > 0)) {
+           return extractAlphabeitFromWords(alienWords);
+        }
+        
+        return null;
+    }
+
+    private List<Character> extractAlphabeitFromWords(String[] alienWords) {
+        /* Get the current order by prefix size, starting from 0 */
+        List<Character> dict = getBasicAlphaBeit(alienWords, 0); /* Basic Alphabeit */ 
+        /* A,R,C from the example! */
+
+        /* Add from prefix while there are letters after first letter of prefix */
+        while (!addToDictByPrefix(alienWords, dict)){} 
+
+        /* Add all the chars we cannot decide in the end*/
+        for (String w : alienWords) {
+            for (Character c: w.toCharArray()) {
+                if (!dict.contains(c)) {
+                    dict.add(c);
+                }
+            }
+        }
+        
+        return dict;
+    }
```

Comment:
Regarding comments, it is important to comment the things that are not obvious, and not comment the things that are obvious. Putting a comment on a constructor saying it's a constructor is not helpful. Having a tricky method not make clear what its effect is supposed to be is also unhelpful. It's much easier to see if a method does what the method comment says it's supposed to do than to try to work out what it does.

I would say a good rule of thumb for method comments is that there should be a clear method comment for any method that it's not obvious what it's doing from a quick glance. I think you are putting those comments in, but I'm not finding them helpful.

I'll add some code review comments to your method comments to try to make it clearer what I'm expecting.

As for bugs, I've found bugs in everyone else's implementation of this assignment. It is tricky. Perhaps we should concentrate on making the code understandable first, and return to looking for bugs later?